src/react/actions/zenko.ts

```diff
@@ -128,12 +135,81 @@ function _getSearchObjects(
   };
 }
 
+function _getSearchVersions(
+  bucketName: string,
+  query: string,
+  keyMarker?: Marker,
+  versionIdMarker?: Marker,
+) {
+  return (dispatch: DispatchFunction, getState: GetStateFunction) => {
+    const { zenkoClient } = getClients(getState());
+    const params = {
+      Bucket: bucketName,
+      Query: query,
+      KeyMarker: keyMarker ? keyMarker : void 0,
+      VersionIdMarker: versionIdMarker ? versionIdMarker : void 0,
+    };
+    dispatch(zenkoClearError());
+    dispatch(networkStart(NETWORK_START_ACTION_SEARCHING_OBJECTS));
+    return zenkoClient
+      .searchBucketVersions(params)
+      .then(
+        async ({
+          NextKeyMarker,
+          NextVersionIdMarker,
+          Version,
+          DeleteMarker,
+          CommonPrefixes,
+          Prefix,
+        }: SearchBucketVersionsResp) => {
+          const list =
+            await _getObjectListWithSignedUrlAndLockStatus<S3Version>(
+              zenkoClient,
+              bucketName,
+              Version,
+            );
+          if (NextKeyMarker) {
+            return dispatch(
+              continueListObjectVersionsSuccess(
```

Comment:
Will this really re-trigger a search if the search contains a lot of items ? I'm afraid not, I think It will try to fetch a second page of objects but using the marker of the search. Likely a bug here

Reply:
Indeed, I think in the end we only need listObjectVersionsSuccess action here.. 
and the continueListObject login will be handle within the table